supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/HtCondorJobExecutionActor.scala

```diff
@@ -136,9 +152,23 @@ class HtCondorJobExecutionActor(override val jobDescriptor: BackendJobDescriptor
   /**
     * Abort a running job.
     */
-  override def abort(): Unit = throw new UnsupportedOperationException("HtCondorBackend currently doesn't support aborting jobs.")
+  override def abort(): Unit = {
+    // Nothing to do in case `condorJobId` is not defined
+    condorJobId foreach { id =>
+      log.info("{}: Aborting job [{}:{}].", tag, jobDescriptor.key.tag, id)
+      val abortProcess = new HtCondorProcess
+      val argv = Seq(HtCondorCommands.Remove, id)
+      val process = abortProcess.externalProcess(argv)
+      val exitVal = process.exitValue()
```

Comment:
I think I still don't understand what is expected here. (I do get the semantics you explained above, and have removed the `AbortedResponse` message that I was sending directly from here before)

The BackendJobExecutionActor has:
```scala
case AbortJobCommand =>
      abort()
      context.parent ! AbortedResponse(jobDescriptor.key)
      context.stop(self)
```
So it expects `abort()` to be called, which when completes, it sends the `AbortedResponse` to the parent.  For Jes and SFS I think we are overirding the default behavior (Jes is extending AsyncBackendJobExecutionActor while SFS overrides it with it's own `receive` definition), which allows for the 3rd point you mentioned above. Currently for this backend (and by default), there is just an `abort()` method which the contract enforces to implement, synchronously (being devoid of any Future[Unit] or expecting any completion msg). Anything I'm missing?